packages/@lwc/jest-preset/types/index.d.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable */
+// Inspired by:
+// - https://unpkg.com/browse/expect@29.6.2/build/index.d.ts
+// - https://unpkg.com/browse/@testing-library/jest-dom@6.0.1/types/matchers.d.ts
+
+import { type expect } from '@jest/globals';
+
+interface LwcJestMatchers<E, R> {
+    /**
+     * @description
+     * Assert that appending a custom element to the DOM causes an error to be thrown from that element's `connectedCallback`.
+     * @example
+     * expect(() => {
+     *     document.body.appendChild(element);
+     * }).toThrowInConnectedCallback('expected error message');
+     * @see
+     * [@lwc/jest-preset docs](https://github.com/salesforce/lwc-test/blob/master/packages/%40lwc/jest-preset/README.md#custom-matchers)
+     */
+    toThrowInConnectedCallback(expected?: unknown): R;
```

Comment:
Why not use the same types as Jest's `toThrow`
```suggestion
    toThrowInConnectedCallback(expected?: string | Constructable | RegExp | Error): R;
```

Reply:
`unknown` is [actually what they use in Jest](https://github.com/jestjs/jest/blob/fb7d95c8af6e0d65a8b65348433d8a0ea0725b5b/packages/expect/src/types.ts#L345-L352).

I'm not sure why they do that though TBH. I'm fine with changing it to be stricter.

Reply:
Also what is `Constructable`? Is it [from `vitest`](https://github.com/vitest-dev/vitest/blob/c03faa2204dc758900d5dbab732af83ee947e4e1/packages/expect/src/types.ts#L135C1-L136)?

Reply:
[Found it in `vitest`](https://github.com/vitest-dev/vitest/blob/dff60e82804ea2eb17aeff6752c1fa53b5f910b8/packages/utils/src/types.ts#L23-L25)

I think overall though I'd rather align with Jest, since otherwise people may run into issues when they change `toThrow()` to `toThrowInConnectedCallback()`.

Reply:
Nevermind, I'm totally wrong; [it's in `@types/jest`](https://unpkg.com/browse/@types/jest@29.5.3/index.d.ts). I'll fix this.

Reply:
Fixed in e0e70bb